File: db/src/main/java/win/whitelife/db/RealmTypes.java
package win.whitelife.db;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author wuzefeng
 * 2018/5/18
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface RealmTypes {

    RealmType value();
    
    
}

File: db/src/main/java/win/whitelife/db/RealmTypes.kt
package win.whitelife.db

import java.lang.annotation.RetentionPolicy
import java.lang.annotation.Retention

/**
 * @author wuzefeng
 * 2018/5/10
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FUNCTION,AnnotationTarget.PROPERTY_GETTER,AnnotationTarget.PROPERTY_SETTER)
annotation class RealmTypes(val value: RealmType)